jax/lax_linalg.py

```diff
@@ -72,6 +72,9 @@ def svd(x, full_matrices=True, compute_uv=True):
 
 def triangular_solve(a, b, left_side=False, lower=False, transpose_a=False,
                      conjugate_a=False, unit_diagonal=False):
+  # TODO(shoyer): remove this hack!
```

Comment:
The need for this suggests that I'm doing something wrong...

Reply:
Zeros shouldn't escape the AD system, i.e. they should only appear in JVP rules. Maybe a check for `ad_util.zero` just needs to be added upstream, i.e. in the caller (assuming this is called from a JVP rule).